qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/settings/settings_model.py

```diff
@@ -26,3 +34,126 @@ def get_setting(name, return_type=str):
+        if use_euler_angles:
+            error_msg = ""
+            euler_scheme = settings["euler_angles_scheme"]
+            euler_sense = settings["euler_angles_sense"]
+            # check sense is comma separated
+            euler_sense_valid, sense_vals, sense_msg = self._validate_euler_sense_string(euler_sense)
+            if not euler_sense_valid:
+                error_msg += sense_msg
+
+            # validate euler_scheme
+            euler_scheme_valid = all([v in ("x", "y", "z") for v in euler_scheme.lower()])
+            if not euler_scheme_valid:
+                error_msg += "Euler Scheme should be defined in terms of X, Y and Z (eg. XYZ or YZY). "
+
+            # check euler_scheme has correct number of corresponding senses
+            sense_scheme_valid = len(euler_scheme) == len(sense_vals)
+            if not sense_scheme_valid:
+                error_msg += "Should be an equal number of Rotation Axes defined as euler senses. "
+            valid = euler_sense_valid and euler_scheme_valid and sense_scheme_valid
+            if not valid:
+                logger.error(error_msg)
+
+    @staticmethod
+    def _validate_euler_sense_string(euler_sense):
+        try:
+            sense_vals = [int(x) for x in euler_sense.split(",")]
+            valid = all([v in (1, -1) for v in sense_vals])
+            error_msg = ""
+        except Exception as e:
+            error_msg = "Euler Senses should be comma separated +/-1s. " + str(e)
+            sense_vals = []
+            valid = False
+        return valid, sense_vals, error_msg
+
+    @staticmethod
+    def _validate_convert_to_float(settings, setting_name):
+        val = ""
+        try:
+            val = settings[setting_name]
+            float(val)
+        except ValueError:
+            logger.error(f"Could not convert {setting_name} value of {val} to a float")
+
+    @staticmethod
+    def validate_path_empty_or_valid(path_to_check):
+        if path_to_check:
+            if not path.exists(path_to_check):
+                return False
+        return True
+
+    def _check_and_populate_with_default(self, name):
+        # note that self.settings and self.default settings should be set from presenter before using this method
+        if name not in self.settings_to_validate or self.settings_to_validate[name] == "":
+            self.settings_to_validate[name] = self.default_settings[name]
+
+    def validate_settings(self, settings, default_settings, all_peaks, set_nullables_to_default=True):
+        # set to class attributes to save duplicating arguments and passing settings dict around too much
+        self.settings_to_validate = settings
+        self.default_settings = default_settings
+        for key in list(self.settings_to_validate):
+            if key not in default_settings.keys():
+                del self.settings_to_validate[key]
+
+        # check and populate calls
+        self._check_and_populate_with_default("rd_name")
```

Comment:
This is quite a lot of lines calling the same function with each setting, I wonder if it's more concise to loop over a list of settings? We could hard code this list, or is there any way to get it from elsewhere?